packages/editor/src/client/blocks/richTextBlock/buttons.tsx

```diff
@@ -0,0 +1,234 @@
+import React, {useState} from 'react'
+import {useTranslation} from 'react-i18next'
+import {Button, ControlLabel, Form, FormControl, FormGroup, Modal} from 'rsuite'
+import {Editor, Transforms, Range} from 'slate'
+import {useSlate} from 'slate-react'
+import {
+  ToolbarIconButtonProps,
+  ToolbarButtonProps,
+  ToolbarButton,
+  ToolbarIconButton
+} from '../../atoms/toolbar'
+import {isFormatActive, toggleFormat} from './editorUtils'
+import {Format, InlineFormat} from './formats'
+
+interface SlateBlockButtonProps extends ToolbarIconButtonProps {
+  readonly format: Format
+}
+
+export function FormatIconButton({icon, format}: SlateBlockButtonProps) {
+  const editor = useSlate()
+
+  return (
+    <ToolbarIconButton
+      icon={icon}
+      active={isFormatActive(editor, format)}
+      onMouseDown={e => {
+        e.preventDefault()
+        toggleFormat(editor, format)
+      }}
+    />
+  )
+}
+
+interface SlateBlockButtonWithChildrenProps extends ToolbarButtonProps {
+  readonly format: Format
+}
+
+export function FormatButtonWithChildren({format, children}: SlateBlockButtonWithChildrenProps) {
+  const editor = useSlate()
+
+  return (
+    <ToolbarButton
+      active={isFormatActive(editor, format)}
+      onMouseDown={e => {
+        e.preventDefault()
+        toggleFormat(editor, format)
+      }}>
+      {children}
+    </ToolbarButton>
+  )
+}
+
+export function LinkFormatButton() {
+  const editor = useSlate()
+  const [isLinkDialogOpen, setLinkDialogOpen] = useState(false)
+  const [selection, setSelection] = useState<Range | null>(null)
+
+  const [title, setTitle] = useState('')
+  const [url, setURL] = useState('')
+
+  const validatedURL = validateURL(url)
+  const isDisabled = !validatedURL
+
+  const {t} = useTranslation()
+
+  return (
+    <>
+      <ToolbarIconButton
+        icon="link"
+        active={isFormatActive(editor, InlineFormat.Link)}
+        onMouseDown={e => {
+          e.preventDefault()
+
+          const nodes = Array.from(
+            Editor.nodes(editor, {
+              at: editor.selection ?? undefined,
+              match: node => node.type === InlineFormat.Link
+            })
+          )
+
+          const tuple = nodes[0]
+
+          if (tuple) {
+            const [node] = tuple
+
+            setTitle((node.title as string) ?? '')
+            setURL((node.url as string) ?? '')
+          } else {
+            setTitle('')
+            setURL('')
+          }
+
+          setSelection(editor.selection)
+          setLinkDialogOpen(true)
+        }}
+      />
+      <Modal show={isLinkDialogOpen} size={'sm'} onHide={() => setLinkDialogOpen(false)}>
+        <Modal.Body>
+          <Form fluid={true}>
+            <FormGroup>
+              <ControlLabel>{t('blocks.richText.link')}</ControlLabel>
+              <FormControl
+                errorMessage={url && !validatedURL ? 'Invalid Link' : undefined}
```

Comment:
Is it possible to translate `Invalid Link` ? E.g. `{t('blocks.richText.invalidLInk')}`

Reply:
yes, will update